### PR TITLE
explain STATUS_DLL_NOT_FOUND

### PR DESCRIPTION
**Pull request by mejrs**
I had someone ask me this today, and googling for it I found various other places where this was asked, with no (good) answers.

### CODE REVIEW

File: guide/src/faq.md [hunk 1]
```diff
@@ -180,3 +180,15 @@ done with the `crate` attribute:
...
+
+## I'm trying to call Python from Rust but I get `STATUS_DLL_NOT_FOUND` or `STATUS_ENTRYPOINT_NOT_FOUND`!
+
+This happens on Windows when linking to the python DLL fails or the wrong one is linked. Your `PATH` variable should contain something like...
+
+```text
+C:\Users\<You>\AppData\Local\Programs\Python\Python310
+```
+
+This directory should have, for Python 3.10, the files `python3.dll` and `python310.dll`. If you have made any changes to your `PATH` or installed things recently, try restarting your shell - any changes will not affect already open shells.
```
Comment (adamreichold) [inline guide/src/faq.md:192]: I also had the converse issue of other programs adding themselves and their internally used DLL to the `PATH` thereby messing things. For example, programs written in Java often include a JRE which can mess up a dedicated installation thoroughly. Or ArcGIS contains a Python installation used for scripts and plug-ins which could place unexpected Python DLL there.

Not sure if this is worth mentioning explicitly here, just thought I'd bring it up for consideration.

File: guide/src/faq.md [hunk 2]
```diff
@@ -180,3 +180,15 @@ done with the `crate` attribute:
...
+
+Failing that, you can use [Dependency Walker](https://www.dependencywalker.com/) to diagnose linking errors.
```
Comment (davidhewitt) [inline guide/src/faq.md:194]: Potentially also worth mentioning `PyOxidizer` as a way to build a self-contained `.exe`?

**Review by davidhewitt — COMMENTED** [review]
Thanks, good idea to document this!

File: guide/src/faq.md (hunk 1 above)
Comment (mejrs) [reply guide/src/faq.md:192]: Yes, I hit this in https://github.com/PyO3/pyo3/issues/2219 😓

**Comment by adamreichold** [comment]
I do think we could bikeshed the wording some more, but I also think this is a definitive improvement over not having this section. So I'll just go ahead and

bors r+

and take the blame if there is any.

**Pull request merged**